WcaOnRails/app/webpacker/components/StaticPages/Logo.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import I18n from '../../lib/i18n';
+import I18nHTMLTranslate from '../I18nHTMLTranslate';
+import '../../stylesheets/static_pages/logo.scss';
+
+/**
+ * @returns {JSX.Element}
+ * @constructor
+ */
+function Logo({ title }) {
+  return (
+    <div className="wca-logo-information">
+      <h1>{title}</h1>
+      {I18n.t('logo.paragraphs').map((_, index) => (
+        <I18nHTMLTranslate
+          // eslint-disable-next-line react/no-array-index-key
+          key={`logo-paragraphs-${index}`}
+          i18nKey={`logo.paragraphs.${index}`}
```

Comment:
In our translation YML, we index paragraphs starting with `1`. I am 99.9999999999% sure that the `.map((_, index))` invocation starts counting `index` at `0`. Have you tried this out locally to make sure that you're not getting any off-by-one errors?

Reply:
Yes I was surprised by this too! As the first element in the array is `empty` (NOT `undefined` or `null`), `map` starts at the first element in the array, not from 0. From the docs:
![image](https://user-images.githubusercontent.com/49319713/187639122-de1ec76e-bff8-4048-ad57-636433704d74.png)
Therefore, as the index of the first element is 1, it starts from 1.


This code:
![image](https://user-images.githubusercontent.com/49319713/187641089-9eaa34fa-5f7d-48d1-b755-67b972c29b49.png)
Logs this:
![image](https://user-images.githubusercontent.com/49319713/187641131-f968c05a-009c-4abf-9ddb-0b2e2341cba2.png)

Reply:
Did some digging and I found it! The library we're using to export Ruby on Rails translations to the JS world is internally doing some stuff with `lodash`. In particular, it uses `_.set` to write the flat key hierarchy to the global `translations` object that is being accessed when you use `I18n.t`.

Try the following code to see what's going on:
```js
// skip this if you already imported lodash
import _ from 'lodash';

var translationsStore = {};
_.set(translationStore, 'foo.bar.1', 'an awesome translated paragraph!');
_.set(translationStore, 'foo.bar.2', 'another awesome translated paragraph!');

console.log(translationsStore.foo.bar);
```

The problem is the Lodash `set` method that interprets our keys as array indices. I don't see how this could be cleverly prevented without changing all of our original `yml` translations to start indexing at zero (which we definitely *don't* want to do right now because it's too much work for this small PR!)

Realistically, I see two options:
1. Change the code to something like `I18n.t('logo.paragraphs').filter(Boolean).map(...foo....)`
2. Hard-code the four paragraphs like in `RegionalOrganizations/index.js`

I tend towards option (2) but I'm open for arguments and debate!

Reply:
Thanks for the research!

I feel like (2) makes it very annoying to change the contents of that page, although as I assume it hasn't changed much since the logo's creation in 2005 that may not be an issue :P

I also like (1) because it feels almost like something you should do regardless when using 'dirty' arrays like this, to make it more resilient to bad yml, say empty lines or other unexpected 'empty' elements of the array, especially while indexing is still from 1.

I did think of some other, less good options:

3. Change the indexing in that one place in the yml to start form 0, as this is the only place it is used.
4. Create a custom translate wrapper which will filter out empty values from arrays.

I also wonder why indexing was every from 1 in the first place? What are the chances that it is due to something with Rails or the Rails i18n library that might break if changed?

Reply:
Indeed (2) is a bit annoying to change / update later on. The reason why we started indexing the YAML at 1 instead of 0 is unclear. I was unable to find Rails documentation that clearly tells you to do so.

Of course, we could just as well use a YML llist. But even changing only the current `logo.paragraphs` key (while there are many more keys that use a similar notion of enumerated "paragraphs", they just haven't been ported over to React yet) would entail changing it in all existing translation YML files (over 20 methinks) which is a lot of tedious work. If you want to go down that path, you should opt for a native array structure like so:

```yaml
logo:
  paragraphs:
    - "some paragraph"
    - "some other paragraph"
    - etc...
  some_other_key:
    foo:
    bar:
```

and see whether that works.

Defining a helper to access paragraphs also sounds like a good idea, see `webpacker/lib/i18n.js` for the current configuration. Monkey-patching a `I18n.paragraphs('some.key')` method should be feasible in JS.

Of course, you can also create the monkey patch _and_ update the YML, but I won't blame you if you don't want to go through the repetitive labor of updating 20+ files :sweat_smile:

Reply:
Ah that is very true I completely forgot you have to update in every locale file...

I like the idea of adding something to the `I18n` global object actually, I'll create a seperate PR for that and you can see if you like that.